lib/request.js

```diff
@@ -357,19 +357,34 @@ AWS.Request = inherit({
    *     request.send();
    */
   send: function send(callback) {
+    this.preventMultipleCalls();
     if (callback) {
       // append to user agent
       this.httpRequest.appendToUserAgent('callback');
       this.on('complete', function (resp) {
         callback.call(resp, resp.error, resp.data);
       });
     }
+    this._didSend = true;
     this.runTo();
 
     return this.response;
   },
 
   /**
+   * @api private
+   */
+  preventMultipleCalls: function preventMultipleCalls() {
```

Comment:
Maybe instead of throwing, this method could take a callback? That would allow the RequestAlreadyTriggeredError to be handled by user callbacks when `send` is called or emitted onto streams when `createReadStream` is called.